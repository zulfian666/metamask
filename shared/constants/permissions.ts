export const CaveatTypes = Object.freeze({
  restrictReturnedAccounts: 'restrictReturnedAccounts' as const,
  restrictNetworkSwitching: 'restrictNetworkSwitching' as const,
});

export const RestrictedEthMethods = Object.freeze({
  eth_accounts: 'eth_accounts',
});

export const RestrictedMethods = Object.freeze({
  eth_accounts: 'eth_accounts',
  snap_dialog: 'snap_dialog',
  snap_notify: 'snap_notify',
  snap_manageState: 'snap_manageState',
  snap_getBip32PublicKey: 'snap_getBip32PublicKey',
  snap_getBip32Entropy: 'snap_getBip32Entropy',
  snap_getBip44Entropy: 'snap_getBip44Entropy',
  snap_getEntropy: 'snap_getEntropy',
  snap_getLocale: 'snap_getLocale',
  wallet_snap: 'wallet_snap',
  ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
  snap_manageAccounts: 'snap_manageAccounts',
  ///: END:ONLY_INCLUDE_IF
} as const);

// ConnectionPermission is pseudo permission used to make possible
// displaying pre-approved connections in the UI seamlessly, alongside other permissions.
export const ConnectionPermission = Object.freeze({
  connection_permission: 'connection_permission',
});

export * from './snaps/permissions';
