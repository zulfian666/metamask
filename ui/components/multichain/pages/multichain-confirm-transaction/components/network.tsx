import React, { useContext } from 'react';
import {
  AlignItems,
  BackgroundColor,
  BorderRadius,
  Display,
  FlexDirection,
  JustifyContent,
  TextColor,
} from '../../../../../helpers/constants/design-system';
import {
  AvatarNetwork,
  AvatarNetworkSize,
  Box,
  Text,
} from '../../../../component-library';
import { I18nContext } from '../../../../../contexts/i18n';
import { MultichainProviderConfig } from '../../../../../../shared/constants/multichain/networks';

export type MultichainTransactionNetworkProps = {
  network: MultichainProviderConfig;
};
export const MultichainTransactionNetwork = ({
  network,
}: MultichainTransactionNetworkProps) => {
  const t = useContext(I18nContext);

  return (
    <Box
      borderRadius={BorderRadius.LG}
      backgroundColor={BackgroundColor.backgroundDefault}
      display={Display.Flex}
      justifyContent={JustifyContent.spaceBetween}
      alignItems={AlignItems.center}
      padding={4}
    >
      <Text color={TextColor.textDefault}>{t('network')}</Text>
      <Box
        display={Display.Flex}
        flexDirection={FlexDirection.Row}
        alignItems={AlignItems.center}
      >
        <AvatarNetwork
          className="mm-picker-network__avatar-network"
          src={network.rpcPrefs?.imageUrl ?? ''}
          name={network.nickname}
          size={AvatarNetworkSize.Xs}
          marginRight={2}
        />
        <Text
          color={TextColor.textDefault}
          data-testid="multichain-confirmation-destination-network"
        >
          {network.nickname}
        </Text>
      </Box>
    </Box>
  );
};
