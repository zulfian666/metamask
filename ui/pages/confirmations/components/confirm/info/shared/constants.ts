export const HEX_ZERO = '0x0';
