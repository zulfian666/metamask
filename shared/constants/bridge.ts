import { CHAIN_IDS } from './network';

// TODO read from feature flags
export const ALLOWED_BRIDGE_CHAIN_IDS = [
  CHAIN_IDS.MAINNET,
  CHAIN_IDS.BSC,
  CHAIN_IDS.POLYGON,
  CHAIN_IDS.ZKSYNC_ERA,
  CHAIN_IDS.AVALANCHE,
  CHAIN_IDS.OPTIMISM,
  CHAIN_IDS.ARBITRUM,
  CHAIN_IDS.LINEA_MAINNET,
  CHAIN_IDS.BASE,
];

export const BRIDGE_DEV_API_BASE_URL = 'https://bridge.dev-api.cx.metamask.io';
export const BRIDGE_PROD_API_BASE_URL = 'https://bridge.api.cx.metamask.io';
export const BRIDGE_API_BASE_URL = process.env.BRIDGE_USE_DEV_APIS
  ? BRIDGE_DEV_API_BASE_URL
  : BRIDGE_PROD_API_BASE_URL;

export const BRIDGE_CLIENT_ID = 'extension';
