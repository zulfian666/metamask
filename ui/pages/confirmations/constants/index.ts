import { CHAINLIST_CHAIN_IDS_MAP } from '../../../../shared/constants/network';

export const EIP712_PRIMARY_TYPE_PERMIT = 'Permit';

export const IGNORE_GAS_LIMIT_CHAIN_IDS = [CHAINLIST_CHAIN_IDS_MAP.MANTLE];

export const TYPED_SIGNATURE_VERSIONS = {
  V1: 'V1',
  V3: 'V3',
  V4: 'V4',
};
