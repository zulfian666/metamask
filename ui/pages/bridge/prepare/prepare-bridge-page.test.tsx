import React from 'react';
import { renderWithProvider } from '../../../../test/jest';
import configureStore from '../../../store/store';
import { createBridgeMockStore } from '../../../../test/jest/mock-store';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import PrepareBridgePage from './prepare-bridge-page';

describe('PrepareBridgePage', () => {
  it('should render the component', () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [],
      },
      {},
      { fromTokenInputValue: 1 },
    );
    const { container, getByRole } = renderWithProvider(
      <PrepareBridgePage />,
      configureStore(mockStore),
    );

    expect(container).toMatchSnapshot();

    expect(getByRole('button', { name: 'ETH' })).toBeInTheDocument();
    expect(getByRole('button', { name: 'USDC' })).toBeInTheDocument();
  });
});
