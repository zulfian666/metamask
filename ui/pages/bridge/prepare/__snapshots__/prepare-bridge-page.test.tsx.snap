// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrepareBridgePage should render the component 1`] = `
<div>
  <div
    class="prepare-bridge-page"
  >
    <div
      class="mm-box prepare-bridge-page__content mm-box--flex-direction-column mm-box--width-full"
    >
      <div
        class="mm-box from-input mm-box--flex-direction-row mm-box--width-full"
      >
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
        >
          Bridge from
        </p>
        0x1:ETH 1
        <div
          class="mm-box multichain-asset-picker"
        >
          <div
            class="multichain-asset-picker__asset"
          >
            <button
              class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-primary mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-inverse mm-box--background-color-primary-default mm-box--rounded-pill"
            >
              ETH
            </button>
          </div>
          <div
            class="multichain-asset-picker__amount"
          />
        </div>
        <div
          class="mm-box mm-text-field mm-text-field--size-md mm-text-field--truncate mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-sm mm-box--border-width-1 box--border-style-solid"
        >
          <input
            autocomplete="off"
            class="mm-box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md mm-box--margin-0 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--color-text-default mm-box--background-color-transparent mm-box--border-style-none"
            focused="false"
            type="text"
            value="1"
          />
        </div>
      </div>
      <div
        class="mm-box to-input mm-box--flex-direction-row mm-box--width-full"
      >
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
        >
          Bridge to
        </p>
        undefined:USDC 0
        <div
          class="mm-box multichain-asset-picker"
        >
          <div
            class="multichain-asset-picker__asset"
          >
            <button
              class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-primary mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-inverse mm-box--background-color-primary-default mm-box--rounded-pill"
            >
              USDC
            </button>
          </div>
          <div
            class="multichain-asset-picker__amount"
          />
        </div>
        <div
          class="mm-box mm-text-field mm-text-field--size-md mm-text-field--truncate mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-sm mm-box--border-width-1 box--border-style-solid"
        >
          <input
            autocomplete="off"
            class="mm-box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md mm-box--margin-0 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--color-text-default mm-box--background-color-transparent mm-box--border-style-none"
            focused="false"
            readonly=""
            type="text"
            value="0"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
